main.go

```diff
@@ -35,23 +38,56 @@ func main() {
-	rootCmd.PersistentFlags().StringToStringVarP(&overriddenLicences, "override-licence", "o", map[string]string{}, "can be used to override the licence detected for a project - e.g. github.com/spf13/cobra=MIT. Repeat this flag to specify multiple values.")
+	rootCmd.PersistentFlags().StringToStringVarP(&overriddenLicences, "override-licence", "o", map[string]string{}, "can be used to override the licence detected for a project directory - e.g. vendor/github.com/spf13/cobra=MIT. Repeat this flag to specify multiple values.")
+	rootCmd.PersistentFlags().StringToStringVarP(&overriddenModuleLicences, "override-module-licence", "m", map[string]string{}, "can be used to override the licence detected for a go module - e.g. github.com/spf13/cobra=MIT. Repeat this flag to specify multiple values.")
 	rootCmd.PersistentFlags().StringSliceVarP(&restrictedLicences, "restricted-licence", "r", []string{}, "licence that will fail the compliance check if found for a project. Repeat this flag to specify multiple values.")
 	rootCmd.PersistentFlags().StringVarP(&logLevel, "log-level", "L", "", "(output) should be one of: (none), debug, info, warn, error, fatal, panic. default (none)")
 	rootCmd.PersistentFlags().BoolVarP(&showComplianceAll, "show-compliance-all", "A", false, "(output) to show compliance checks as JSON regardless of outcome")
 	rootCmd.PersistentFlags().BoolVarP(&showComplianceErrors, "show-compliance-errors", "E", false, "(output) to show compliance checks as JSON only in case of errors")
+	rootCmd.PersistentFlags().BoolVarP(&checkGoModules, "check-go-modules", "", false, "check all go modules a project depends on")
 	rootCmd.MarkPersistentFlagRequired("restricted-licence")
 }
 
 func validateCompliance(_ *cobra.Command, args []string) {
 	setLogLevel(logLevel)
 
+	if len(overriddenModuleLicences) > 0 && len(overriddenLicences) > 0 {
+		logAndExit("Only use one of --override-module-licence (%d uses) and --override-licence (%d uses)", len(overriddenModuleLicences), len(overriddenLicences))
+	}
+
+	for module, licence := range overriddenModuleLicences {
+		cmd := exec.Command("go", "list", "-m", "-f", "\"{{.Dir}}\"", module)
+
+		out, err := cmd.Output()
+		if err != nil {
+			logAndExit("Failed to list go module %s: output: %s, error: %s", module, out, err)
+		}
+		pkgDir := strings.Trim(strings.TrimSpace(string(out)), "\"")
+		overriddenLicences[pkgDir] = licence
+	}
+
 	config := compliance.Config{
 		RestrictedLicences:        restrictedLicences,
 		IgnoredProjects:           ignoredProjects,
 		OverriddenProjectLicences: overriddenLicences,
 	}
 
+	if checkGoModules {
+		if len(args) > 0 {
+			logAndExit("--check-go-modules and positional args cannot be set at the same time (received %d)", len(args))
+		}
+
+		var err error
+		args, err = getGoModules()
+		if err != nil {
```

Comment:
I would add a log.Info here to print the modules found. 
This would then be displayed when using `-L INFO`